toolchain/language_server/handle_shutdown.cpp

```diff
@@ -0,0 +1,16 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/language_server/handle.h"
+
+namespace Carbon::LanguageServer {
+
+auto HandleShutdown(
+    Context& /*context*/,
+    const clang::clangd::NoParams& /*client_capabilities*/,
+    llvm::function_ref<void(llvm::Expected<std::nullptr_t>)> on_done) -> void {
+  on_done(nullptr);
```

Comment:
Looking at the `shutdown` and `exit` spec, I think there are a few missing details here that it'd be good to have a TODO to flesh out:

```suggestion
  // TODO: Track that `shutdown` was called, and:
  // - Warn on duplicate calls.
  // - Make `exit` return `1` if `shutdown` wasn't called.
  //   https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#exit
  // - Error on other post-`shutdown` calls.
  on_done(nullptr);
```

Reply:
Done.